### PR TITLE
Add `matchesSelector()` assertions

### PR DESCRIPTION
**Pull request by lindem**
working at #206 please comment

### CODE REVIEW

File: lib/assertions/matches-selector.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export default function matchesSelector(
+  targetSelector: string | Element,
+  compareSelector: string
+): Array<number> {
+  let targetElements: Array<Element> =
+    typeof targetSelector === 'string'
+      ? Array.from(document.querySelectorAll(targetSelector))
+      : [targetSelector];
```
Comment (Turbo87) [inline lib/assertions/matches-selector.ts]: this code should probably reuse the `findElements()` method of the `Assertions` class. I also think we should just pass an `Element[]` array to this function, and let the surrounding code worry about dealing with the target selectors.

File: lib/assertions/matches-selector.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  let failures = targetElements.reduce(function(acc, element) {
+    return element.matches(compareSelector) ? acc : acc + 1;
+  }, 0);
```
Comment (Turbo87) [inline lib/assertions/matches-selector.ts]:
```js
let failures = targetElement.filter(it => it.matches(compareSelector)).length > 0;
```
👆 seems a little easier to read to me

File: lib/__tests__/matches-selector.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      expect(assert.results).toEqual([{
+        actual: 2,
+        expected: 3,
+        message: '1 out of 3 elements selected by p did not also match the selector p + p.',
+        result: false
+      }]);
+    });
+  })
+
+  describe('doesNotMatchSelector: success scenarios', () => {
```
Comment (Turbo87) [inline lib/__tests__/matches-selector.ts]: `doesNotMatchSelector` should have it's own dedicated testing file and not be nested in the `assert.dom(...).matchesSelector()` test suite

File: lib/__tests__/matches-selector.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      assert.dom('p').matchesSelector('div>p');
+
+      expect(assert.results).toEqual([{
+        actual: 3,
+        expected: 3,
+        message: '3 elements, selected by p, also match the selector div>p.',
+        result: true
+      }]);
+    });
+  });
```
Comment (Turbo87) [inline lib/__tests__/matches-selector.ts]: this is missing test cases where an `Element` is passed directly to `assert.dom(...)`. have a look at the other test suites for examples.

Comment (lindem) [reply lib/__tests__/matches-selector.ts]: So I did accomodate instances of Element passed to the assertion and looked at how hasClass does it.

File: lib/__tests__/matches-selector.ts (hunk 3 above)
Comment (lindem) [reply lib/__tests__/matches-selector.ts]: I placed these tests into another file and also reduced redundancy in wording.

File: lib/assertions/matches-selector.ts (hunk 2 above)
Comment (lindem) [reply lib/assertions/matches-selector.ts]: agreed, but to count failures, we filter the negative results, making it `!it.matches(...)`.

File: lib/assertions/matches-selector.ts (hunk 1 above)
Comment (lindem) [reply lib/assertions/matches-selector.ts]: I applied the suggested changes.

**Comment by lindem** [comment]
Done with it again (for now). I am unsure if the new redundancy in the assertions.ts file is warranted (doesNotMatchSelector and matchesSelector have a lot in common there), but it works and is easy to understand I hope.

Still happy to continue working on it. Thanks for patience.

File: lib/__tests__/does-not-match-selector.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    document.body.innerHTML = '<div><p class="first"></p><p></p><p class="last"></p></div>';
+  });
+
+  describe('success scenarios', () => {
+    test('one element (selector passed) not matching compareSelector', () => {
+      assert.dom('p.first').doesNotMatchSelector('p + p');
+
+      expect(assert.results).toEqual([{
+        actual: 0,
+        expected: 0,
```
Comment (Turbo87) [inline lib/__tests__/does-not-match-selector.ts]: we should use better `actual/expected` values. have a look at some of the other assertions, we often reuse the assertion message, and generate another assertion message for `expected`.

Comment (lindem) [reply lib/__tests__/does-not-match-selector.ts]: I have been coming down with severe health problems during the weekend. I will get to it once I am well enough to code again. Sorry for the delay.

Comment (Turbo87) [reply lib/__tests__/does-not-match-selector.ts]: no worries, gute besserung 😉

**Comment by lindem** [comment]
@Turbo87 I hope this is more what you wanted. I am seemingly unable to resolve the merge conflict in this PR myself (it is not clear that I *can* do something about it from the page the button leads to, but then, I cannot write to your repository).

Ah, and happy new year.

File: lib/__tests__/does-not-match-selector.ts (hunk 5 above)
Comment (lindem) [reply lib/__tests__/does-not-match-selector.ts]: thanks very much.

**Pull request merged**